### ISSUE TITLE
Manifest not updated for 1.4.0, so Obsidian is installing 1.3.0

### ISSUE THREAD

**Issue by claremacrae** [KraXen72/obsidian-emoji-autocomplete#3]
In https://github.com/phibr0/obsidian-emoji-shortcodes/issues/47#issuecomment-2021590508, I thought I was testing the performance of 1.4.0 - only to find after the fact that 1.3.0 was installed.

Here is why:

<img width="1397" alt="image" src="https://github.com/KraXen72/obsidian-emoji-autocomplete/assets/4840096/584c5b71-7e24-4bcc-9e54-ddce2eac2048">

**Comment by claremacrae** [comment]
Obsidian uses the version number in the manifest.json on the default branch to decide which version to install for users.

So when a release is done, the version in the manifest on the default branch needs to be updated, so users get the update.

**Comment by KraXen72** [comment]
ohh, my bad . i ran the version bump command though, and it didn't bump it??

**Comment by KraXen72** [comment]
i think there is some git wierdness because i pushed some commits for the 1.3.0 => 1.4.0 update as well and they're not showing up?

**Comment by KraXen72** [comment]
i downloaded the manifest from the 1.4.0 release and it has 1.4.0. could it be the versions.json file?
![Screenshot_20240327-000453_Brave.png](https://github.com/KraXen72/obsidian-emoji-autocomplete/assets/21956756/f56d7e50-b8e7-42fe-8acb-e12373de902d)

**Comment by claremacrae** [comment]
> Obsidian uses the version number in the manifest.json on the default branch to decide which version to install for users.
> 
> So when a release is done, the version in the manifest on the default branch needs to be updated, so users get the update.


Did you see the above comment? That explains what is happening - if you look at the screenshot are shared, it confirms the problem.

What Obsidian will do is to go this this URL - note the `master` - and get the version number:
https://github.com/KraXen72/obsidian-emoji-autocomplete/blob/master/manifest.json

And that file currently looks like this:

<img width="1124" alt="image" src="https://github.com/KraXen72/obsidian-emoji-autocomplete/assets/4840096/dcc47ba5-6bf6-447d-9e3f-4a10e1b2504d">

So Obsidian says "ah right, 1.3.0 is the current release, that's what I will serve to users".

To put it another way, although the 1.4.0 release was created yesterday, the master branch does not contain any changes since the 1.3.0 release was created on 21st Feb.

**One solution would be to merge to `master` the changes you put in the 1.4.0 release.**

I suggest asking for help in #plugin-dev on Discord if you need more info... I don't have any other ways to explain this...

**Comment by claremacrae** [comment]
> i think there is some git wierdness because i pushed some commits for the 1.3.0 => 1.4.0 update as well and they're not showing up?

Where did you push those changes to? They are not on this repo:
https://github.com/KraXen72/obsidian-emoji-autocomplete

**Comment by claremacrae** [comment]
Here are some links on doing releases:

- Manually
    - https://docs.obsidian.md/Plugins/Releasing/Submit+your+plugin#Step+2+Create+a+release
- With automation - see the notes at the end
    - https://docs.obsidian.md/Plugins/Releasing/Release+your+plugin+with+GitHub+Actions

**Comment by KraXen72** [comment]
sorry, i misread your comment. i understand. i'll push a fixed manifest and changes to master later.

**Comment by KraXen72** [comment]
okay it should show 1.4.1 now. Can you just quickly see if the percievable performance is any better? i'll investigate it further if it's not satisfactory. Thank you for you work and time so far.

**Comment by claremacrae** [comment]
Thanks. I'm unlikely to have time to try it out again for several days, but will try at some point.

Weirdly, Obsidian now says it was updated to 1.4.1 two days ago - but it does at least show that it has picked up the update.

![image](https://github.com/KraXen72/obsidian-emoji-autocomplete/assets/4840096/651e3a9c-6687-4b09-874a-b6d9c8bd5824)

And 1.4.1 installs fine too.

So I think you can close this issue.

**Comment by KraXen72** [comment]
btw i figured out what was the issue
- i use git to backup my vault
- the git repo for this plugin was recognized as a submodule, and whenever i backed up the vault it reverted to 1.3.0, throwing away all my commits. 
- i restored 1.4.0's source code from vs code's local edit history
- i added the git repo for this plugin to the vault's gitignore and removed the submodule.
- there was also some issue with submodules and git reset `<commit hash>`,  they are not friends
